scheduler/util.go

```diff
@@ -442,6 +457,93 @@ func networkPortMap(n *structs.NetworkResource) map[string]int {
 	return m
 }
 
+func affinitiesUpdated(jobA, jobB *structs.Job, taskGroup string) bool {
+	var aAffinities []*structs.Affinity
+	var bAffinities []*structs.Affinity
+
+	tgA := jobA.LookupTaskGroup(taskGroup)
+	tgB := jobB.LookupTaskGroup(taskGroup)
+
+	// Append jobA job and task group level affinities
+	aAffinities = append(aAffinities, jobA.Affinities...)
+	aAffinities = append(aAffinities, tgA.Affinities...)
```

Comment:
This will panic if the taskgroup was added in jobB but didn't exist in jobA. (tgA will be nil). Should guard access to tgA behind a nil check

Worth adding a test case for as well.

Reply:
Determined that before this method is called we short circuit if task group was removed (https://github.com/hashicorp/nomad/blob/c346a47b5b3db306c60a8a598f20dd13c5599db9/scheduler/reconcile.go#L317) 

Worth adding a comment for future readers of this code though

Reply:
comment added